server.go

```diff
@@ -2549,6 +2551,117 @@ func setupRPCListeners() ([]net.Listener, error) {
 	return listeners, nil
 }
 
+// miningAddrStore encapsulates a set of coinbase generation addresses in a
+// concurrent safe interface. All addresses stored MUST be unique, any
+// duplicates are rejected.
+type miningAddrStore struct {
+	miningAddrs []btcutil.Address
+	sync.RWMutex
+}
```

Comment:
w.r.t my comment earlier about using an interface instead, `miningAddrStore` has little to do with the rest of what is happening in `server.go` and would be better suited in the `mining/cpuminer` package.